tests/test_dia_collection.py

```diff
@@ -0,0 +1,172 @@
+    """ Edit the center coordinate and id of a source record in place.
+
+    Parameters
+    ----------
+    src : lsst.afw.table.SourceRecord
+        Input SourceRecord to edit.
+    id : int
+        Integer value to set the record id to.
+    ra_degrees : float
+        RA value to set the record coordinate to.
+    dec_degrees : float
+        DEC value to set the record coordinate to.
+    scatter_arcsec : float
+        Arcsecond scatter to add to the position of the source record coord.
+    """
+    coord = Coord(afwGeom.Angle(ra_degrees, units=afwGeom.degrees),
+                  afwGeom.Angle(dec_degrees, units=afwGeom.degrees))
+    if scatter_arcsec > 0.0:
+        coord.offset(
+            afwGeom.Angle(np.random.rand() * 360, units=afwGeom.degrees),
+            afwGeom.Angle(np.random.rand() * scatter_arcsec,
+                          units=afwGeom.arcseconds))
+    src.setCoord(coord)
+    src['id'] = src_id
+
+
+class TestDIAObjectCollection(unittest.TestCase):
+
+    def test_init(self):
+        """ Test that we can properly create a DIAObjectCollection from a list
+        of DIAObjects.
+        """
+        obj_list = create_test_dia_objects(1, 1, 0.1)
+        obj_collection = DIAObjectCollection(obj_list)
+
+        self.assertTrue(obj_collection.is_updated)
+        self.assertTrue(obj_collection.is_valid_tree)
+
+        self.assertEqual(obj_collection.get_dia_object_ids()[0], 0)
+        self.assertEqual(obj_collection.get_dia_object(0).get('id'), 0)
+
+    def test_append_and_update(self):
+        """ Test that we can add a new DIAObject to an existing
+        DIAObjectCollection.
+        """
+        obj_list = create_test_dia_objects(1, 1, 0.1)
+        obj_collection = DIAObjectCollection(obj_list)
+
+        new_dia_obj = create_test_dia_objects(1, 1, 0.1, 1, 0.1)[0]
+        obj_collection.append(new_dia_obj)
+        self.assertFalse(obj_collection.is_updated)
+        self.assertFalse(obj_collection.is_valid_tree)
+
+        obj_collection.update_dia_objects()
+        self.assertTrue(obj_collection.is_updated)
+        obj_collection.update_spatial_tree()
+        self.assertTrue(obj_collection.is_valid_tree)
+
+        self.assertEqual(len(obj_collection.get_dia_object_ids()), 2)
+        self.assertEqual(obj_collection.get_dia_object(1).get('id'), 1)
+
+    def test_score_and_match(self):
+        """ Test association between a set of sources and an existing
+        DIAObjectCollection.
+
+        This also tests that a DIASource that can't be associated within
+        tolerance is addpended to the DIAObjectCollection as a new
+        DIAObject.
+        """
+        obj_collection = DIAObjectCollection(
+            create_test_dia_objects(4, 1, -1))
+        src_cat = create_test_dia_sources(5)
+        for src_idx, src in enumerate(src_cat):
+            edit_source_record(
+                src, src_idx + 4, 0.1 * src_idx, 0.1 * src_idx, -1)
```

Comment:
Again, if you're going to split the args on new lines, do them all (and you shouldn't have to for this one)